var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -117,6 +119,21 @@ class Hdf5(AutotoolsPackage):
     patch('h5public-skip-mpicxx.patch', when='@:1.8.21,1.10.0:1.10.5+mpi~cxx',
           sha256='b61e2f058964ad85be6ee5ecea10080bf79e73f83ff88d1fa4b602d00209da9c')
 
+    # The argument 'buf_size' of the C function 'h5fget_file_image_c' is declared as intent(in) though
+    # it is modified by the invocation. As a result, aggressive compilers such as Fujitsu's may do a
+    # wrong optimization to cause an error.
+    def patch(self):
+        if os.path.exists('fortran/src/H5Fff.F90'):
+            filter_file(
+                'INTEGER\(SIZE_T\), INTENT\(IN\) :: buf_size',
+                'INTEGER(SIZE_T), INTENT(OUT) :: buf_size',
+                'fortran/src/H5Fff.F90')
```

Comment:
```suggestion
                'fortran/src/H5Fff.F90', string=True)
```

Reply:
@adamjstewart I've modified the code according to your suggestions.